Move instance list pagination and selection text above table

edlerd opened the pull request:
## Done

- moved instance list pagination above the table
- moved instance selection notification into the pagination component

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - check instance table selection -- also with pagination

webteam-app commented:
Demo starting at https://lxd-ui-475.demos.haus

lorumic commented on src/components/SelectedTableNotification.tsx:
```diff
@@ -0,0 +1,72 @@
+  };
+
+  return (
+    <Notification borderless className="u-no-margin--bottom" title="Selection">
```
I think that we said the first row (the tile) is superfluous. I think it should be removed as it occupies one row of precious space.
```suggestion
    <Notification borderless className="u-no-margin--bottom">
```

lorumic reviewed (commented):
The pagination controls jump up and down when toggling a selection:

![Peek 2023-09-29 09-45](https://github.com/canonical/lxd-ui/assets/56583786/ddcc3d5d-c948-4b9e-848e-1b81493ea58c)

lorumic reviewed (approved):
Thanks for applying the change. Just one other small improvement that I thought of, but feel free to implement it or not.

edlerd commented on src/components/SelectedTableNotification.tsx:
```diff
@@ -0,0 +1,72 @@
+  setSelectedNames,
+}: Props) => {
+  const isAllSelected =
+    selectedNames.length === filteredNames.length && selectedNames.length > 0;
```
Good idea to relax the check here, I'll change that.

We need to check in the parent, because in the pagination component we would still receive a component even if the component renders null. But in the pagination component, we want to show the amount of items, when nothing is selected. Thus, we want to receive a null in the no selection case.

piperdeck commented:
I think I've settled on the idea that we discussed in our standup on..... Tuesday I think. I propose doing the following:

- move the pagination bar into the header above the table
- When the selection information appears, it does so inside the pagination bar, replacing the "Showing X out of X instances" message

Figma link: https://www.figma.com/file/1b1Qjh8raA55aWx7gbccM7/LXD-Improvements?type=design&node-id=249-99&mode=design

piperdeck reviewed (changes requested):
See comment

edlerd commented:
> I think I've settled on the idea that we discussed in our standup on..... Tuesday I think. I propose doing the following:
> 
> * move the pagination bar into the header above the table
> * When the selection information appears, it does so inside the pagination bar, replacing the "Showing X out of X instances" message

Moved it up and made the content the same height.
I removed the notification icon, because that caused the selected info to break into 3 or more rows on smaller screen widths.

piperdeck reviewed (approved):
This looks great!
